azure-client-runtime/src/main/java/com/microsoft/azure/v2/AzureProxy.java

```diff
@@ -141,84 +139,83 @@ protected Object handleAsyncHttpResponse(final HttpRequest httpRequest, Single<H
-                        final String originalHttpRequestMethod = originalHttpRequest.httpMethod();
-
-                        PollStrategy pollStrategy = null;
-                        if (httpStatusCode == 200) {
-                            pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, expectsResourceResponse, serializer, delayInMilliseconds);
-                            if (pollStrategy != null) {
-                                result = Single.just(pollStrategy);
-                            }
-                            else {
-                                result = createProvisioningStateOrCompletedPollStrategy(originalHttpRequest, originalHttpResponse, methodParser, delayInMilliseconds);
-                            }
-                        }
-                        else if (originalHttpRequestMethod.equalsIgnoreCase("PUT") || originalHttpRequestMethod.equalsIgnoreCase("PATCH")) {
-                            if (httpStatusCode == 201) {
-                                pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, expectsResourceResponse, serializer, delayInMilliseconds);
-                                if (pollStrategy == null) {
-                                    result = createProvisioningStateOrCompletedPollStrategy(originalHttpRequest, originalHttpResponse, methodParser, delayInMilliseconds);
-                                }
-                            } else if (httpStatusCode == 202) {
-                                pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, expectsResourceResponse, serializer, delayInMilliseconds);
-                                if (pollStrategy == null) {
-                                    pollStrategy = LocationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, delayInMilliseconds);
-                                }
-                            }
-                        }
-                        else {
-                            if (httpStatusCode == 202) {
-                                pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, expectsResourceResponse, serializer, delayInMilliseconds);
-                                if (pollStrategy == null) {
-                                    pollStrategy = LocationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, delayInMilliseconds);
-                                    if (pollStrategy == null) {
-                                        throw new CloudException("Response does not contain an Azure-AsyncOperation or Location header.", originalHttpResponse);
+                        final int[] longRunningOperationStatusCodes = new int[] {200, 201, 202};
```

Comment:
This is good-- now manually adding extra status codes is no longer necessary or desired in the generator.